samples: i2s_codec: Add and update support for various NXP devices

VitekST opened the pull request:
Add support for the following targets:
- `mimxrt1170_evk/mimxrt1176/cm4`
- `rd_rw612_bga`
- `lpcxpresso55s69/lpc55s69/cpu0`

Also fix pin mapping and Flexcomm signal sharing on the `mimxrt685_evk/mimxrt685s/cm33`.

VitekST committed:
dts: mimxrt685_evk: Map all pins of Flexcomm #1

Map all pins of RT685's Flexcomm #1 peripheral.

Needed for opration of Flexcomm #1 in I2S mode or Flexcomm #3 in I2S
mode with shared signals.

Signed-off-by: Vit Stanicek <vit.stanicek@nxp.com>

VitekST committed:
samples: i2s_codec: Enable on lpcxpresso55s69/cpu0

Modify board's pinctrl nodes to include the MCLK signal. Instantiate the
wm8904 driver. Add board specific Kconfig fragment and DT overlay. Set
up the MCLK signal.

Signed-off-by: Vit Stanicek <vit.stanicek@nxp.com>

VitekST committed:
samples: i2s_codec: Enable on rd_rw612_bga

Instantiate the wm8904 driver in rd_rw612_bga's DT. Add MCLK clock
configuration to rw612's soc.c. Modify pinmux definitions to route
those. Add DT and Kconfig fragments to the sample. Document output jack
utilised.

Signed-off-by: Vit Stanicek <vit.stanicek@nxp.com>

VitekST committed:
samples: i2s_codec: Enable on mimxrt1170_evk@B/cm4

Add aliases to domain DT overlay and instantiate the wm8962 driver in
it. Add Kconfig fragment and a DT overlay for the sample.

Signed-off-by: Vit Stanicek <vit.stanicek@nxp.com>

VitekST committed:
boards: mimxrt685_evk/cm33: Update I2S sharing

Add SCK and WS signals to a shared signal set between Flexcomm #1 and
Flexcomm #3 for the mimxrt685_evk/mimxrt685s/cm33.

This enables the board to both transmit and receive audio, as the BCK
and WS signals produced by the WM8904 codec are only connected to the
Flexcomm #1 peripheral.

Signed-off-by: Vit Stanicek <vit.stanicek@nxp.com>

hakehuang reviewed (commented):
@VitekST , please update the i2s_code sample.yaml for new added platforms

hakehuang commented:
I did not get any output from earphone port on mimxrt1170_evk

hakehuang commented on samples/drivers/i2s/i2s_codec/sample.yaml:
```diff
@@ -6,8 +6,11 @@ tests:
+      - mimxrt1170_evk@B/mimxrt1176/cm4
       - mimxrt700_evk/mimxrt798s/cm33_cpu0
       - mimxrt1060_evk/mimxrt1062/qspi
+      - lpcxpresso55s69/cpu0
```
should be 
```
lpcxpresso55s69/lpc55s69/cpu0
```

hakehuang commented:
a build fail for cm4
sample.drivers.i2s.output on mimxrt1170_evk@B/mimxrt1176/cm4 error (Build failure - error: static assertion failed: "NUM_DMA_BLOCKS_RX_PREP must be < CONFIG_DMA_TCD_QUEUE_SIZE")

VitekST commented:
@hakehuang I know, but that's outside the scope of this PR, that build failure is about the `output` sample, not the `i2s_codec` demo. Honestly, the `mimxrt1170_evk` targets shouldn't be even among the targets applicable to that sample.

hakehuang commented:
> @hakehuang I'm aware, it's just that, on multiple occasions, I was told not to increase scopes of my PRs.

 I would say, for this time, it is necessary, because if we run git bisect this PR introduce the failure.

hakehuang commented:
> sample for the `mimxrt1170_evk/mimxrt1176/cm4` on, let's say, `209bfbde897` (currently the HEAD of `main`), it fails already. This PR doesn't do anything with the `output` demo, it just happens to be based on a revision where that target is selected to be built, still before the commits I've pushed here.
@VitekST, please take a look at the sample.yaml filter
```
  sample.drivers.i2s.output:
    filter: dt_alias_exists("i2s-tx")
```
as you enable the i2x-tx in m4 board in this pr, the CI will include this sample for testing.

VitekST commented:
@hakehuang Oh! Apologies, thanks for clarifying that. I didn't intend to do that. This sample is dependent on a different alias, so I'll just remove the `i2s-tx` one.

VitekST commented:
@hakehuang Thanks for clarifying. Point taken.

dleach02 commented:
Use a separate commit in this PR to clean up the compliance issue. 

The Zephyr Project recently added this to CI and unfortunately it doesn't seem to know how to deal with patchsets. So what contributors have been doing is just fixing the file within the PR as a separate commit with a commit message of something "cleaning up compliance".

VitekST committed:
dts: lpcxpresso55s69: Make mikrobus_i2c compliant

Remove non-compliant EOL in the mikrobus_i2c empty node declaration in
boards/nxp/lpcxpresso55s69/lpcxpresso55s69.dtsi.

Signed-off-by: Vit Stanicek <vit.stanicek@nxp.com>
